src/shogun/features/SparseFeatures.cpp

```diff
@@ -474,9 +476,11 @@ template<class ST> float64_t CSparseFeatures<ST>::dense_dot(int32_t vec_idx1, co
 			"sparse_matrix[%d] check failed (matrix features %d >= vector dimension %d)\n",
 			vec_idx1, get_num_features(), sv.get_num_dimensions());
 
-		REQUIRE(vec2_len >= sv.get_num_dimensions(),
-			"sparse_matrix[%d] check failed (dense vector dimension %d >= vector dimension %d)\n",
-			vec_idx1, vec2_len, sv.get_num_dimensions());
+		REQUIRE(
```

Comment:
Putting these check so low level is costly. Why not check once outside the loop instead? Not sure.....

Reply:
I agree that they're more costly, but I think putting checks in slightly lower level functions is cleaner and less error prone (like Linalg does).